### PR TITLE
Added an HttpLoggingInterceptor into the OkHttpClient

### PR DESCRIPTION
**Pull request by elektron9**
Added an HttpLoggingInterceptor into the OkHttpClient, which will log raw requests and responses at the INFO level.

### CODE REVIEW

**Comment by palantirtech** [comment]
Thanks for your interest in [palantir/http-remoting](https://github.com/palantir/http-remoting), @elektron9! Before we can accept your pull request, you need to sign our contributor license agreement - just visit https://cla.palantir.com/ and follow the instructions. Once you sign, I'll automatically update this pull request.

File: retrofit2-clients/src/main/java/com/palantir/remoting1/retrofit2/Retrofit2ClientBuilder.java [hunk 1]
```diff
@@ -101,6 +102,11 @@ public Request authenticate(Route route, Response response) throws IOException {
             client.addInterceptor(new RetryInterceptor(config.maxNumRetries()));
         }
 
+        // HTTP request/response logging
+        HttpLoggingInterceptor httpLoggingInterceptor = new HttpLoggingInterceptor();
+        httpLoggingInterceptor.setLevel(HttpLoggingInterceptor.Level.BODY);
```
Comment (uschi2000) [inline retrofit2-clients/src/main/java/com/palantir/remoting1/retrofit2/Retrofit2ClientBuilder.java]: This is not OK as a default.
(1) Bodies can be huge, they shouldn't be logged in a default configuration
(2) Log-level should probably be TRACE, not INFO

Comment (elektron9) [reply retrofit2-clients/src/main/java/com/palantir/remoting1/retrofit2/Retrofit2ClientBuilder.java]: Good call. Adding configuration settings and changing the log level to `TRACE`

**Comment by elektron9** [comment]
@uschi2000 - I realized that both the `jaxrs-clients` and `retrofit2-clients` made use of `OkHttp` for logging, so I added the configuration to both of those projects. Let me know what you think about this, thanks!

File: http-clients-api/src/main/java/com/palantir/remoting1/clients/ClientConfig.java [hunk 2]
```diff
@@ -81,5 +88,5 @@ public static Builder builder() {
         return new Builder();
     }
 
-    public static final class Builder extends ImmutableClientConfig.Builder {}
+    public static final class Builder extends com.palantir.remoting1.clients.ImmutableClientConfig.Builder {}
```
Comment (uschi2000) [inline http-clients-api/src/main/java/com/palantir/remoting1/clients/ClientConfig.java]: don't qualify

File: jaxrs-clients/src/main/java/com/palantir/remoting1/jaxrs/FeignJaxRsClientBuilder.java [hunk 3]
```diff
@@ -69,9 +69,14 @@
 import okhttp3.Credentials;
 import okhttp3.Response;
 import okhttp3.Route;
+import okhttp3.logging.HttpLoggingInterceptor;
+import org.slf4j.LoggerFactory;
 
 public final class FeignJaxRsClientBuilder extends ClientBuilder {
 
+
+    private static final org.slf4j.Logger logger = LoggerFactory.getLogger(FeignJaxRsClientBuilder.class);
```
Comment (uschi2000) [inline jaxrs-clients/src/main/java/com/palantir/remoting1/jaxrs/FeignJaxRsClientBuilder.java]: don't qualify

**Comment by uschi2000** [comment]
looks ok modulo two nitpicks. thanks!

**Comment by elektron9** [comment]
@uschi2000 - all set!

**Comment by uschi2000** [comment]
Closing due to security concerns: we don't want to give products the option of logging all traffic in production.

**Pull request closed without merging**